duckduckgo/iOS
# first party url logic

Reviewer: Mia
Asana: https://app.asana.com/0/392891325557410/433193852567931
CC:

**Description**:

* Use TLDs to determine if the domains of two URLs match in order to check that a URL is loaded from the same domain or not.

* Updated the logic so that all URLs are processed every time (instead of caching what happened to them previously.

* Load the scripts even if content blocking is disabled in settings.

**Steps to test this PR**:

Test 1 - First Party URLs
1. Go to criteo.com and ensure the site works as expected

Test 2 - blocking setting
1. Disable content blocking in settings
2. Visit a tracker heavy site
3. Ensure tracker detection messages are being received by the app


###### Reviewer Checklist:


###### PR DRI Checklist:
